libyul/ObjectParser.cpp

```diff
@@ -104,9 +110,80 @@ shared_ptr<Block> ObjectParser::parseCode()
 	return parseBlock();
 }
 
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping() const
+{
+	return tryGetSourceLocationMapping(
+		m_scanner->currentCommentLiteral(),
+		m_scanner->currentCommentLocation(),
+		m_errorReporter
+	);
+}
+
+optional<ObjectParser::ReverseSourceNameMap> ObjectParser::tryGetSourceLocationMapping(
+	std::string const& _text,
+	SourceLocation const& _location,
+	ErrorReporter& _errorReporter
+)
+{
+	// @use-src 0:"abc.sol", 1:"foo.sol", 2:"bar.sol"
+	//
+	// UseSrcList := UseSrc (',' UseSrc)*
+	// UseSrc     := [0-9]+ ':' FileName
+	// FileName   := "(([^\"]|\.)*)"
+
+	// Matches some "@use-src TEXT".
+	static std::regex const lineRE = std::regex(
+		R"~~~((^|\s+)@use-src\s+(.*)$)~~~",
+		std::regex_constants::ECMAScript | std::regex_constants::optimize
+	);
+	std::smatch sm;
+	if (!std::regex_search(_text, sm, lineRE))
+		return nullopt;
+	solAssert(sm.size() == 3, "");
+
+	// Let @c text point to the parameter value (last match).
+	auto text = sm[2].str();
+
+	// iteratively match for NUM : STRING_LITERAL and increment
+	static regex const firstParamRE(R"~~~(\s*(\d+)\s*:\s*"((?:\\\"|[^\"])*)")~~~",
```

Comment:
What about `"\\""`? Maybe we need a real scanner?

Reply:
Yeah, just quickly constructing a `Scanner` here is probably the easiest idea...

Reply:
done.